internal/setup/setup.go

```diff
@@ -176,35 +177,40 @@ func (r *Release) validate() error {
 						}
 						return fmt.Errorf("slices %s and %s conflict on %s", old, new, newPath)
 					}
-					// Note: We do not have to record newPath because conflict
-					// is a transitive relation.
-					continue
-				}
-
-				// Check for glob and generate conflicts.
-				for oldPath, old := range paths {
-					oldInfo := old.Contents[oldPath]
-					if !(newInfo.Kind == GlobPath || newInfo.Kind == GeneratePath ||
-						oldInfo.Kind == GlobPath || oldInfo.Kind == GeneratePath) {
-						continue
-					}
-					if (newInfo.Kind == GlobPath || newInfo.Kind == CopyPath) &&
-						(oldInfo.Kind == GlobPath || oldInfo.Kind == CopyPath) {
-						if new.Package == old.Package {
-							continue
-						}
-					}
-					if strdist.GlobPath(newPath, oldPath) {
-						if (old.Package > new.Package) || (old.Package == new.Package && old.Name > new.Name) ||
-							(old.Package == new.Package && old.Name == new.Name && oldPath > newPath) {
-							old, new = new, old
-							oldPath, newPath = newPath, oldPath
-						}
-						return fmt.Errorf("slices %s and %s conflict on %s and %s", old, new, oldPath, newPath)
+					// Note: Because for conflict resolution we only check that
+					// the created file would be the same and we know newInfo and
+					// oldInfo produce the same one, we do not have to record
+					// newInfo.
+				} else {
+					paths[newPath] = new
+					if newInfo.Kind == GeneratePath || newInfo.Kind == GlobPath {
+						conflictable[newPath] = new
```

Comment:
The term "globs" still feels more adequate here, even if we have GeneratePath on it.  The point of that code path is to run these through GlobPath. "conflictables" could be anything.

Reply:
Agreed.